tests/servers/test_basehttp.py

```diff
@@ -106,6 +107,55 @@ def makefile(mode, *a, **kw):
 
         self.assertEqual(body, b"HTTP_SOME_HEADER:good")
 
+    def test_no_body_returned_for_head_requests(self):
+        hello_world_body = b"<!DOCTYPE html><html><body>Hello World</body></html>"
+
+        def test_app(environ, start_response):
+            """A WSGI app that returns a hello world"""
+            start_response("200 OK", [])
+            http_environ_items = sorted(
+                "%s:%s" % (k, v) for k, v in environ.items() if k.startswith("HTTP_")
+            )
+            yield (",".join(http_environ_items)).encode()
+            yield hello_world_body
+
+        # when making a GET request, the body is returned
+        rfile = BytesIO(b"GET / HTTP/1.0\r\n")
+        rfile.seek(0)
+
+        wfile = UnclosableBytesIO()
+
+        def makefile(mode, *a, **kw):
+            if mode == "rb":
+                return rfile
+            elif mode == "wb":
+                return wfile
+
+        request = Stub(makefile=makefile)
+        server = Stub(base_environ={}, get_app=lambda: test_app)
+
+        # Prevent logging from appearing in test output.
+        with self.assertLogs("django.server", "INFO"):
+            # instantiating a handler runs the request as side effect
+            WSGIRequestHandler(request, "192.168.0.2", server)
+
+        wfile.seek(0)
+        body = list(wfile.readlines())[-1]
+        # confirm that the body is returned in a GET response
```

Comment:
To add more explanation, comments should state the expected behavior and omit prefixes like "Confirm that" since all tests confirm things. This guideline is from [Python coding style](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style).

```suggestion
        # The body is returned in a GET response.
```
and 
```python
        # The body is not returned in a HEAD response.
```